src/network/networkctl.c

```diff
@@ -486,6 +497,18 @@ static int acquire_link_info(sd_bus *bus, sd_netlink *rtnl, char **patterns, Lin
                 c++;
         }
 
+        /* Look if we matched all our arguments that are not globs. It
+         * is OK for a glob to match nothing, but not for an exact argument. */
+        for (size_t pos = 0; pos < strv_length(patterns); pos++)
+                if (!matched_patterns[pos]) {
```

Comment:
bad memory access if the new0() above failed. Also, in order to keep the indentation levels small, I'd reverse the check here and `continue` if it doesn't match.